streams-tests/shared/src/test/scala/zio/stream/ZSinkSpec.scala

```diff
@@ -167,12 +167,34 @@ object ZSinkSpec extends ZIOBaseSpec {
             }(equalTo(Chunk(Right(3), Left("Aie"))))
           )
         ),
-        test("dropWhile")(
-          assertZIO(
-            ZStream(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
-              .pipeThrough(ZSink.dropWhile[Int](_ < 3))
-              .runCollect
-          )(equalTo(Chunk(3, 4, 5, 1, 2, 3, 4, 5)))
+        suite("dropWhile")(
+          test("happy path")(
+            assertZIO(
+              ZStream(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
+                .pipeThrough(ZSink.dropWhile[Int](_ < 3))
+                .runCollect
+            )(equalTo(Chunk(3, 4, 5, 1, 2, 3, 4, 5)))
+          ),
+          test("correct leftovers behavior") {
+            assertZIO(
+              ZStream
+                .range(0, 20, chunkSize = 3)
+                .run(ZSink.dropWhile[Int](_ <= 10).collectLeftover)
+                .map(_._2)
+            )(equalTo(Chunk(11)))
+          },
+          test("combined with head, transduced") {
+            assertZIO(
+              ZStream(0, 0, 0, 1, 0, 0, 2)
+                .rechunk(3)
+                .transduce {
+                  ZSink.dropWhile[Int](_ == 0) *> ZSink.head[Int]
+                }
+                .take(10)
+                .runCollect
+              //seems like the 'trailing' sink invocation is by design
+            )(equalTo(Chunk(Some(1), Some(2), None)))
```

Comment:
I feel like the correct behaviour here should be `Chunk(Some(1))` since we have `ZSink.head[Int]`?

Reply:
`transduce` (based on `ZStream.fromSink`) runs the sink repeatedly until upstream completes. each time the sink completes, its result is emitted and the cycle begins again.

in this specific example:
first invocation of the sink consumes `0, 0, 0, 1` and emits `Some(1)`,
second invocation consumes `0, 0, 2` and emits `Some(2)`,
third invocation operates on an empty stream and emits `None`, then the entire stream completes.